### PR TITLE
[CELEBORN-750] Simplify get Netty used direct memory logic

### PR DESCRIPTION
**Pull request by cfmcgrady**
### What changes were proposed in this pull request?

netty has exposed the public API `PlatformDependent.usedDirectMemory()` to get netty used direct memory since [netty-4.1.35.Final](https://github.com/netty/netty/pull/8945), simplifies the logic

### Why are the changes needed?

simplifies the get netty used direct memory logic

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Pass GA

### CODE REVIEW

**Review by pan3793 — APPROVED** [review]

**Review by FMX — APPROVED** [review]
LGTM. Thanks.

File: worker/src/main/java/org/apache/celeborn/service/deploy/worker/memory/MemoryManager.java [hunk 1]
```diff
@@ -207,7 +203,7 @@ private MemoryManager(CelebornConf conf) {
         () ->
             logger.info(
                 "Direct memory usage: {}/{}, disk buffer size: {}, sort memory size: {}, read buffer size: {}",
-                Utils.bytesToString(nettyMemoryCounter.get()),
+                Utils.bytesToString(PlatformDependent.usedDirectMemory()),
```
Comment (FMX) [inline worker/src/main/java/org/apache/celeborn/service/deploy/worker/memory/MemoryManager.java:206]: I think there needs a check to make sure that Celeborn Worker has been configured with off-heap memory. Or Celeborn will encounter undefined behavior.

Comment (FMX) [reply worker/src/main/java/org/apache/celeborn/service/deploy/worker/memory/MemoryManager.java:206]: I think adding an assertion about  "PlatformDependent.usedDirectMemory() != -1 " could be enough.

Comment (cfmcgrady) [reply worker/src/main/java/org/apache/celeborn/service/deploy/worker/memory/MemoryManager.java]: make sense, addressed.

**Pull request closed without merging**